htdocs/core/lib/functions.lib.php

```diff
@@ -7587,7 +7587,7 @@ function natural_search($fields, $value, $mode = 0, $nofirstand = 0)
 				$i3 = 0;
 				foreach($tmpcrits as $tmpcrit)
 				{
-					if(empty($tmpcrit)) continue;
+					if(!isset($tmpcrit)) continue;
```

Comment:
If $tmpcrit is set to '', this will generate problem i think. 
So a test empty() or === '0' seems better.